plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -228,6 +228,14 @@ public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
         }
     }
 
+    @Override
+    public void dropCorruptedTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        // Since it is currently not possible to obtain the table location, even if we drop the table from the metastore,
+        // it is still impossible to delete the table location.
+        throw new TrinoException(NOT_SUPPORTED, "dropCorruptedTable is not supported for Iceberg REST catalogs");
```

Comment:
`"dropCorruptedTable is not supported for Iceberg REST catalogs"` -> "Cannot drop corrupted table %s from Iceberg REST catalog"

This is intended to be a more end-user friendly message.